### ISSUE TITLE
idea: update process hooks

### ISSUE THREAD

**Issue by jahkeup** [bottlerocket-os/bottlerocket#560]
# Idea Proposal: Update Process Hooks

***tl;dr** - I think we should add hooks in the update process to apply quick fixes at update time.*

The process of updating a host involves several components of the OS to fully write out and migrate a host, any issue that prevents this process from succeeding will prevent later updates from happening as the current host is usedto apply these updates. During the course of the OS's lifetime - [only once as of now](#558 ) - we need to be able to address a situation where an update requires host action (from the un-updated current host!) that it would otherwise be unequipped to take and specifically outside the context of migrations. 

To work around any unforeseeable issues we must already have in place mechanisms on the updating host to "hotfix" itself (or the process) during the update at critical checkpoints. I propose that we implement "hooks" listed in the update manifest that are assigned "checkpoints" throughout the process. These would be predefined and integrated into the various components involved to allow for flexibly addressing unknown circumstances. As these hooks would be list in the update manifest the security assertions are maintained for these blobs as well. I think that providing the means to protect users from bugs in the update process is critical and should be done ASAP as support for these hooks has to be already on the host.

cc: @zmrow @iliana

**Comment by jpmcb** [comment]
Closing. Not currently planned to implement this functionality.
